src/pyimpspec/data/formats/par.py

```diff
@@ -0,0 +1,91 @@
+# pyimpspec is licensed under the GPLv3 or later (https://www.gnu.org/licenses/gpl-3.0.html).
+# Copyright 2023 pyimpspec developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# The licenses of pyimpspec's dependencies and/or sources of portions of code are included in
+# the LICENSES folder.
+
+from os.path import exists
+from typing import (
+    IO,
+    List,
+)
+from pyimpspec.data.data_set import (
+    DataSet,
+    dataframe_to_data_sets,
+)
+
+
+def parse_par(path: str) -> List[DataSet]:
+    """
+    Parse a Parstat .par file containing an impedance spectrum.
+
+    Parameters
+    ----------
+    path: str
+        The path to the file to process.
+
+    Returns
+    -------
+    List[DataSet]
+    """
+    from pandas import DataFrame
+    assert isinstance(path, str) and exists(path), path
+    fp: IO
+    with open(path, "r", encoding="latin1") as fp:
+        lines: List[str] = list(
+            filter(lambda _: _ != "", map(str.lower, map(str.strip, fp.readlines())))
+        )
+
+    assert len(lines) > 0, f"Failed to find any impedance data in '{path}'"  # check for blank file
+    # Need to make sure we have the correct Action for the EIS experiment
+    found_eis = False
+    eis_action_no = None
+    while not found_eis:
+        line: str = lines.pop(0)
+        if line.startswith("<action"):
+            eis_action_no = str(int(line[7:][:-1]) - 1)  # Parstat's action segment numbering is off by one for some reason.
+        if "potentiostatic eis" in line:
```

Comment:
What about galvanostatic EIS?

Do you know if the column headers found in the `Segment` sections are the same across different measurement types? It might not be necessary to parse each `Action` section if the contents of a `Segment` section is enough to determine whether or not the results correspond to a potentiostatic/galvanostatic EIS measurement.

For example, it might be enough to check if the `Z Real` and `Z Imag` columns exist and that their corresponding values are not all zero.

Reply:
That's a good thought. My personal analysis functions use other information from the action sections, which may have sample area, reference electrode info, etc. I'll do a little further investigation here, it may be sufficient for pyimpspec purposes to simply look for non-zero values in those columns.